### DISCUSSION TITLE
Sounds like an output midi lost a content input style.

### DISCUSSION THREAD

**Discussion by moonsh** [cifkao/groove2groove#16] [Q&A] [answered]
I need an output that should maintain some of the content input style.
So I can tell it's from the content.
However, after style transfer, it's difficult to catch the content input style from the output.

Is there a way to conserve input music like 40 ~ 50 percent and add a different style?

Maybe this is the limitation of the current model?
I found that you added content input to an output midi again at the last stage in the demo site.

**Comment by cifkao** [comment, marked as answer]
Normally only the harmony is preserved from the content input. Neither style nor the melody of the content input is preserved. This is why I'm adding the content melody to the output.

To achieve what you want, you could try to interpolate the styles of both songs as described in section VI.E of the paper. You would need to feed *both* songs to the style encoder to obtain two style embeddings `s1` and `s2` and then "blend" them e.g. with `s = 0.4 * s1 + 0.6 * s2` and feed this `s` to the decoder as the new style embedding. But note that this still won't allow you to preserve any specific parts of the content input, e.g. there will still be no melody. All the model will try to do is come up with an accompaniment style that is in between the two.

**Reply by moonsh** [reply]
I see. Thanks a lot. I misunderstood the output of the groove2groove. 
Now it's clear.
